sdk/core/azure-core-amqp/inc/azure/core/amqp/network/sasl_transport.hpp

```diff
@@ -8,22 +8,39 @@
 
 namespace Azure { namespace Core { namespace Amqp { namespace Network { namespace _internal {
 
-  class SaslTransport final : public Transport {
+  /** @brief Factory to create a transport using SASL authentication.
+   *
+   */
+  class SaslTransportFactory final {
```

Comment:
Same comment as above about public constructors and destructors in a "static class", `unique_ptr` return type and a necessity of a factory class existence.

Reply:
I'm at a loss here. My version of the sources doesn't have constructors or destructors on the factories - the factories are pure static classes with a static "Create" method.

Reply:
If they don't have any as `private` or `=delete` (or both), they do have them implicit auto-generated. And as a user I may create an instance of such factory, which is not the intended use.